engine-spring/core/src/test/java/org/camunda/bpm/engine/spring/test/SpringProcessEngineInitializationTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.spring.test;
+
+import org.camunda.bpm.engine.ProcessEngine;
+import org.camunda.bpm.engine.ProcessEngines;
+import org.camunda.bpm.engine.impl.util.ReflectUtil;
+import org.camunda.bpm.engine.spring.SpringConfigurationHelper;
+import org.junit.Test;
+
+import java.net.URL;
+
+import static org.junit.Assert.*;
+
+public class SpringProcessEngineInitializationTest {
+
+  @Test
+  public void testSpringConfigurationContextInitializing() {
+    //given
+    URL resource = existActivitiContext();
+    //when
+    ProcessEngine processEngine = SpringConfigurationHelper.buildProcessEngine(resource);
+    //then
+    assertNotNull(processEngine);
+    assertEquals("activitiContextProcessName", processEngine.getName());
+
+    processEngine.close();
+  }
+
+  @Test
+  public void testProcessEnginesInitializingByActivitiContext() {
+    //given
+    existActivitiContext();
```

Comment:
I believe we can take that as a given. If it's not there, the test will fail and we know we broke something by removing the `activiti-context.xml`.
```suggestion
    //given an activiti-context.xml is present on the classpath
```

Reply:
Yes, you're right but I think we need keep this check. Because if we delete this check and remove `activiti-context.xml` from classpath then when we want to get the process engine we get `null`, and the root cause of this may be hidden in many parts of code. Problem may hide in storage, initializing and other. And this check says "this test needs `activiti-context.xml` in the classpath and this test not work without this file".
In `ProcessEngines 114:122` we can see this. If `activiti-context.xml` not exists in a classpath, `ProcessEngines` will just skip this part.

Reply:
I agree, that is a good argument. 👍